From f6ed1bbb31e9ace4fea083b251a57cc2299248bd Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sat, 4 Apr 2020 10:55:25 +0200
Subject: [PATCH 1/1] efi_loader: PlatformLangCodes must be read only

The variable PlatformLangCodes has to be read only according to the UeFI
spec.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_setup.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/efi_loader/efi_setup.c b/lib/efi_loader/efi_setup.c
index 9806aa6584..e5adeb8c8d 100644
--- a/lib/efi_loader/efi_setup.c
+++ b/lib/efi_loader/efi_setup.c
@@ -44,7 +44,8 @@ static efi_status_t efi_init_platform_lang(void)
 	ret = efi_set_variable_int(L"PlatformLangCodes",
 				   &efi_global_variable_guid,
 				   EFI_VARIABLE_BOOTSERVICE_ACCESS |
-				   EFI_VARIABLE_RUNTIME_ACCESS,
+				   EFI_VARIABLE_RUNTIME_ACCESS |
+				   EFI_VARIABLE_READ_ONLY,
 				   sizeof(CONFIG_EFI_PLATFORM_LANG_CODES),
 				   CONFIG_EFI_PLATFORM_LANG_CODES);
 	if (ret != EFI_SUCCESS)
-- 
2.25.1

